tests/Radio.test.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
+import { Radio } from "../lib/components";
+import { render } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+const ControlledRadio = () => {
+  const [value, setValue] = useState("");
+  
+  return (
+    <Radio
+      label="Radio"
+      onChange={(e) => setValue(e.target.value)}
+      value={value}
+    >
+      <Radio.Item label="option1" name="options" value="option1" />
+      <Radio.Item label="option2" name="options" value="option2" />
+    </Radio>
+  );
+};
+
+describe("Radio", () => {
+  it("should render without error", () => {
+    const { getByText } = render(
+      <Radio label="Radio">
+        <Radio.Item label="option1" name="options" value="option1" />
+      </Radio>
+    );
+    expect(getByText("Radio")).toBeInTheDocument();
+  });
+
+  it("should call onChange when radio value is changed", () => {
+    const onChange = jest.fn();
+    const { getByText } = render(
+      <Radio label="Radio" onChange={onChange}>
+        <Radio.Item label="option1" name="options" value="option1" />
+      </Radio>
+    );
+    userEvent.click(getByText("option1"));
+    expect(onChange).toHaveBeenCalledTimes(1);
```

Comment:
@ajmaln Do we have to test what value is getting returned on onChange here? Can we test that? (Doubt)

Reply:
@amaldinesh7 the arguments can be tested through `expect(onChange).toHaveBeenCalledWith()`. However, the `target` property will be null for the events due to synthetic event nullification. 

I found a way by including the assertion in the mock function itself.

```js
it("should call onChange when radio value is changed", () => {
    const onChange = jest.fn((event) => {
      expect(event.target.value).toBe("option1");
    });
    const { getByText } = render(
      <Radio label="Radio" onChange={onChange}>
        <Radio.Item label="option1" name="options" value="option1" />
      </Radio>
    );
    userEvent.click(getByText("option1"));
    expect(onChange).toHaveBeenCalledTimes(1);
});
```

This seems to be working as expected and fails when the assertion value is wrong. Let's do this for testing values.